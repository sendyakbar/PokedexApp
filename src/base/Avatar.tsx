import React from 'react';
import {StyleSheet, Image, ImageStyle} from 'react-native';

export default function Avatar(): JSX.Element {
  return (
    <Image
      source={require('../assets/images/pokeball.png')}
      style={styles.pokeball}
    />
  );
}

type StyleType = {
  pokeball: ImageStyle;
};

const styles = StyleSheet.create<StyleType>({
  pokeball: {
    height: 36,
    width: 36,
  },
});
