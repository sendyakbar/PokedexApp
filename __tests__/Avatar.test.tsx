import 'react-native';
import React from 'react';
import {it} from '@jest/globals';
import renderer from 'react-test-renderer';

import Avatar from '../src/base/Avatar';

it('renders correctly', () => {
  const tree = renderer.create(<Avatar />).toJSON();
  expect(tree).toMatchSnapshot();
});
