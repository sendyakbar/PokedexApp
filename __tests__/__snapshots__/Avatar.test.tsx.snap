// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  source={
    {
      "testUri": "../../../src/assets/images/pokeball.png",
    }
  }
  style={
    {
      "height": 36,
      "width": 36,
    }
  }
/>
`;
